superduper/backends/mongodb/query.py

```diff
@@ -583,9 +598,21 @@ def _execute(self, parent, method='encode'):
         limit_args, _ = self._get_method_parameters('limit')
         limit = {"$limit": limit_args[0]} if limit_args else None
 
-        predict_ids = self.parts[-1][1]
+        outputs_parts = [p for p in self.parts if p[0] == 'outputs']
+        predict_ids = sum([p[1] for p in outputs_parts], ())
 
         pipeline = []
+        filter_mapping_base, filter_mapping_outptus = self._get_filter_mapping()
+        if filter_mapping_base:
+            pipeline.append({"$match": filter_mapping_base})
+            project.update({k: 1 for k in filter_mapping_base.keys()})
+
+        predict_ids_in_filter = [
+            key.replace("_outputs__", "") for key in filter_mapping_outptus.keys()
```

Comment:
Why this `_outputs__`?

Reply:
Optimized the logic by using predict_id.

Additionally, since `MongoMock` does not support ‘.’ in the ‘as’ field for lookups, the key was converted to ‘outputs_’.

After the join, the complete outputs data can be queried as `_outputs__{predict_id}._outputs.{predict_id}` : result.